### PR TITLE
Chore/security fix/validate format file

### PR DESCRIPTION
**Pull request by Isalafont**

### CODE REVIEW

**Comment by Isalafont** [comment]
[fix : valider le format des fichiers "expression de besoin" (pour l'api impot particulier) et "maquette projet"](https://trello.com/c/R4whbLZd/752-fix-valider-le-format-des-fichiers-expression-de-besoin-pour-lapi-impot-particulier-et-maquette-projet)

**Review by rdubigny — CHANGES_REQUESTED** [review]
Cool PR!

Have you find a way to use the spreadsheed docs you added in `backend/spec/fixtures/` in the tests ?

File: backend/app/services/magic_spreadsheet_validator.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# from https://kgrz.io/robust-file-validations-in-ruby.html
+class MagicSpreadsheetValidator
```
[the remark this answers is not in the document]
Comment (Isalafont) [reply backend/app/services/magic_spreadsheet_validator.rb]: Done :)

File: backend/app/services/magic_spreadsheet_validator.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  def read_trailing_signature!
+    @file.rewind
+    @file.seek(max_trailing_signature_bytes * -1, IO::SEEK_END)
+    trailing_bytes = @file.read
+    @trailing_signature = trailing_bytes.unpack1("H*")
+  end
+
+  def starting_signature_valid?
+    @starting_signature.start_with?(*VALID_STARTING_SIGNATURE)
```
[the remark this answers is not in the document]
Comment (Isalafont) [reply backend/app/services/magic_spreadsheet_validator.rb]: `magic_spreadsheet_validator.rb` will inherite now from `magic_validator.rb,`

File: backend/app/models/document/geo_shape.rb [hunk 3]
```diff
@@ -1,8 +1,2 @@
-class Document::GeoShape < Document
-  mount_uploader :attachment, DocumentUploader
-
-  private
-
-  def content_type_validation
-  end
+class Document::GeoShape < PdfDocument
```
[the remark this answers is not in the document]
Comment (Isalafont) [reply backend/app/models/document/geo_shape.rb]: Done

**Comment by Isalafont** [comment]
@rdubigny I'm using them in the document_spec.rb to test the file validations.

**Review by rdubigny — APPROVED** [review]

**Pull request merged**